### ISSUE TITLE
Not work after last update of the extension (waiting on php cs fixer to finish pre..)

### ISSUE THREAD

**Issue by yarovikov** [biati-digital/nova-php-cs-fixer#24]
Hi there! I have fresh problem at this morning. Thanks :)

Version 4.2.0
Nova 9.2
Mac OS 12.1

<img width="507" alt="Screenshot 2022-04-14 at 10 45 23" src="https://user-images.githubusercontent.com/30932012/163338431-df65520b-950e-4bc3-9dbc-603833e9ec75.png">

**Comment by biati-digital** [comment]
Hi, i'm not able to replicate it, are you still having this issue? if so please use the bug report template so you can provide all the info we need.

**Comment by yarovikov** [comment]
@biati-digital hi, 

maybe this 'll help you 

<img width="595" alt="Screenshot 2022-04-15 at 22 26 24" src="https://user-images.githubusercontent.com/30932012/163623699-919d6cba-fdbf-4fd8-a29b-e2616c2402b1.png">

<img width="1345" alt="Screenshot 2022-04-15 at 22 27 51" src="https://user-images.githubusercontent.com/30932012/163623827-cc25f341-69d5-408b-8bcf-753434844143.png">

**Comment by biati-digital** [comment]
Weird, the error says it can not find the php-cs-fixer binary. Please try uninstalling the extension and installing it again.

**Comment by yarovikov** [comment]
@biati-digital the problem still remains :(  it was happend after update to Version 4.2.0
 
<img width="1230" alt="Screenshot 2022-04-16 at 09 27 37" src="https://user-images.githubusercontent.com/30932012/163664500-e5100985-29d5-4f54-adc6-a1e7f7d3b3f1.png">

**Comment by dheliker** [comment]
Same problem for me as well.  Normally there is a PHP process running (the formatter), that process is no longer running.

**Comment by macisbetter** [comment]
Same issues as above.  Thank you!

**Comment by biati-digital** [comment]
Still not able to replicate, i'm on Nova 8.4, my license expired a few months ago and not planning to renew it anytime soon. 
I uninstalled and installed the extension again, once installed i saw a similar error but complaining about Prettier and the format on save not working, i had to disable Prettier and then the formatter started working again.

As is said before, i do not have enough information to make tests or help you. The screenshot only shows partial config, the most important part "PHP path and PHP CS Fixer path" is not displayed in the screenshot, i do not know your PHP version, etc.

It can also be a problem related to the Nova version, i'm not able to test in the latest version as like i said, my license already expired. If that's the case someone will need to help with this.

**Comment by yarovikov** [comment]
@biati-digital you can reset trial like this https://gist.github.com/thoaster/ca5b4fdb653b9cdb3930897045b78558 :)

**Comment by biati-digital** [comment]
@yarovikov that also removes all the app configuration and servers, we have more than 50 servers saved on the app, reset is not an option, i'm not willing to jeopardize our configuration or even invest time trying exporting, importing servers, etc.

**Comment by yarovikov** [comment]
@biati-digital i deleted this file manually, all servers stay. Just are removed editor options like sizes, colors

**Comment by biati-digital** [comment]
@yarovikov  Like i said, not something i'm going to do.

**Comment by yarovikov** [comment]
Hello there again. I found a solution. Need to delete .nova folder from your project

**Comment by biati-digital** [comment]
@yarovikov thank's for letting me know. Hopefully that will help other users who are having this problem.

**Comment by emilorol** [comment]
@biati-digital please check your mailbox for an upgrade solution.

**Comment by emilorol** [comment]
I did some code review and I submitted a PR #25 that make the fix command work again with V3 (3.4.0 & 3.8.0), but I am still experiencing the error described above.

```
Could not open input file: "/Users/admin/Library/Application Support/Nova/Extensions/biati.phpcsfixer/php/php-cs-fixer-3.4.0"
```

Manual execution of command from terminal

```sh
$ php -d display_errors=on "/Users/admin/Library/Application Support/Nova/Extensions/biati.phpcsfixer/php/php-cs-fixer-3.4.0" fix -v "/Users/admin/Library/Application Support/Nova/Workspaces/6BF18F43-1904-47FD-859B-AE2F757E4001/biati.phpcsfixer/publichtmlnewwebsitecomwpcontentthemesbootscorechildcustomwidgets.php" --config="/Users/admin/Library/Application Support/Nova/Extensions/biati.phpcsfixer/rules/wordpress.php_cs" --using-cache=yes --cache-file=/Users/admin/Library/Application\ Support/Nova/Extensions/biati.phpcsfixer/php/.php-cs-fixer.cache
PHP CS Fixer 3.4.0 Si! by Fabien Potencier and Dariusz Ruminski (47177af)
Runtime: PHP 7.4.12
Loaded config default from "/Users/admin/Library/Application Support/Nova/Extensions/biati.phpcsfixer/rules/wordpress.php_cs".
Using cache file "/Users/admin/Library/Application Support/Nova/Extensions/biati.phpcsfixer/php/.php-cs-fixer.cache".
.                                                                                                                                                                                                                         1 / 1 (100%)
Legend: ?-unknown, I-invalid file syntax (file ignored), S-skipped (cached or empty file), .-no changes, F-fixed, E-error

Fixed all files in 0.164 seconds, 6.000 MB memory used
```
Note: I added the `php` flag to display errors `-d display_errors=on` as well as the `fix` flag to display the output in a verbose mode `-v`

**Comment by newelement** [comment]
Any way I can help @emilorol ?

**Comment by newelement** [comment]
I get the same error if I use my local cs fixer path `/usr/local/bin/php-cs-fixer`

**Comment by Rocketpilot** [comment]
> Still not able to replicate, i'm on Nova 8.4, my license expired a few months ago and not planning to renew it anytime soon. I uninstalled and installed the extension again, once installed i saw a similar error but complaining about Prettier and the format on save not working, i had to disable Prettier and then the formatter started working again.

Sorry just to clarify - you no longer have access to current versions of Nova but you're still offering the nova-php-cs-fixer extension to users? It probably should just be pulled, surely?

**Comment by biati-digital** [comment]
@Rocketpilot so you say that i should just remove weeks and weeks of work? there's no option to remove a submitted extension and also the fix should be easy, you know you can send a PR correct? 

If you are not willing to invest your time then i'm afraid that you'll have to wait, i'm having a lot of work and the extension is not a priority. i also created 2 other extensions "Text tools and Advanced New File" i'm not planning to abandon the extensions, i invested a lot of time on them to simply remove them. Hopefully Nova becomes a decent editor in a couple of years.

**Comment by Rocketpilot** [comment]
Look I'm not trying to be confrontational here, but the tool simply doesn't work! I appreciate that you've invested time into it but right now it's not fit for purpose and as you've made clear that you're not a current user of the host application, I don't see that it's reasonable to pretend to users that it is.

**Comment by biati-digital** [comment]
@Rocketpilot the tool works for me in an older Nova version and have worked since the first time it was created, Nova keeps updating and changing things, it's normal that at some point something will break. 

I said that i'm no longer using Nova and that's true, i never said that the development of the extension will stop, you are assuming that and without a reason as the last update was just 3 weeks ago (i stopped using Nova months ago), do you see how you saying that makes absolutely no sense? and instead of saying "just delete it" you could help in some way, at least providing info, logs, etc. i've asked this many times but nobody used the "bug report" template to provide meaningful information. It's just sad and upsetting to read comments like yours.

Thanks to @emilorol who sponsored the update for us and we can keep working on the extension so i'll work on an update.

Again, you are welcome to help in any way you can if not then please do not reply anymore and just wait for the update.

**Comment by biati-digital** [comment]
btw i already sent an updated version to our sponsor, if someone is interested to try it you are welcome to send me an email (manuel [at] biati.digital ) and i'll send a link to download the updated extension, please only do this if you can help providing logs and more info in case you have problems.

**Comment by biati-digital** [comment]
The new update has been released "4.2.1", you can update directly from the extensions gallery, if you have any problems please create an issue following the "bug report" template
